internal/testgen/gen_negative_positive.go

```diff
@@ -0,0 +1,232 @@
+package main
+
+import (
+	"strings"
+	"text/template"
+
+	"github.com/Antonboom/testifylint/internal/checkers"
+)
+
+type NegativePositiveTestsGenerator struct{}
+
+func (NegativePositiveTestsGenerator) Checker() checkers.Checker {
+	return checkers.NewNegativePositive()
+}
+
+func (g NegativePositiveTestsGenerator) TemplateData() any {
+	var (
+		checker = g.Checker().Name()
+		report  = checker + ": use %s.%s"
+	)
+
+	return struct {
+		CheckerName       CheckerName
+		InvalidAssertions []Assertion
+		ValidAssertions   []Assertion
+		IgnoredAssertions []Assertion
+	}{
+		CheckerName: CheckerName(checker),
+		InvalidAssertions: []Assertion{
+			{Fn: "Greater", Argsf: "a, 0", ReportMsgf: report, ProposedFn: "Positive", ProposedArgsf: "a"},
+			{Fn: "Less", Argsf: "0, a", ReportMsgf: report, ProposedFn: "Positive", ProposedArgsf: "a"},
+			{Fn: "True", Argsf: "a > 0", ReportMsgf: report, ProposedFn: "Positive", ProposedArgsf: "a"},
+			{Fn: "True", Argsf: "0 < a", ReportMsgf: report, ProposedFn: "Positive", ProposedArgsf: "a"},
+			{Fn: "False", Argsf: "a <= 0", ReportMsgf: report, ProposedFn: "Positive", ProposedArgsf: "a"},
+			{Fn: "False", Argsf: "0 >= a", ReportMsgf: report, ProposedFn: "Positive", ProposedArgsf: "a"},
+
+			{Fn: "Less", Argsf: "a, 0", ReportMsgf: report, ProposedFn: "Negative", ProposedArgsf: "a"},
+			{Fn: "Greater", Argsf: "0, a", ReportMsgf: report, ProposedFn: "Negative", ProposedArgsf: "a"},
+			{Fn: "True", Argsf: "a < 0", ReportMsgf: report, ProposedFn: "Negative", ProposedArgsf: "a"},
+			{Fn: "True", Argsf: "0 > a", ReportMsgf: report, ProposedFn: "Negative", ProposedArgsf: "a"},
+			{Fn: "False", Argsf: "a >= 0", ReportMsgf: report, ProposedFn: "Negative", ProposedArgsf: "a"},
+			{Fn: "False", Argsf: "0 <= a", ReportMsgf: report, ProposedFn: "Negative", ProposedArgsf: "a"},
+		},
+		ValidAssertions: []Assertion{
+			{Fn: "Positive", Argsf: "a"},
+			{Fn: "Negative", Argsf: "a"},
+		},
+		IgnoredAssertions: []Assertion{
```

Comment:
significantly more tests
check the border cases -1, 0, 1 (I like your idea) for any comparison